lib/src/features/game/domain/game.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:dartchess/dartchess.dart';
+
+part 'game.freezed.dart';
+part 'game.g.dart';
+
+@Freezed(toJson: false)
+class Game with _$Game {
+  factory Game({
+    required String id,
```

Comment:
One thing I'm doing a lot in lila these days is getting reads of `String`s, `Int`s, and other meaningless types. For type safety purpose, I much prefer handling a `GameId` type than a `String`, because then I know, and the compiler knows, what it is.

Is there a nice way to achieve that in dart, ideally without incurring runtime costs?

Reply:
Maybe this is what I'm looking for: https://github.com/dart-lang/language/blob/master/working/1426-extension-types/feature-specification-views.md

Reply:
I suppose we can do it for very critical types (game id) and wait for dart to implement views for others (wrapping with a class comes with a perf penalty)